experiments/AMIP/moist_mpi_earth/coupler_driver.jl

```diff
@@ -0,0 +1,272 @@
+            )
+        dummmy_remap!(z0m_S, combined_field)
+        parent(combined_field) .=
+            combine_surface.(
+                parent(mask),
+                parent(bucket_sim.params.z_0b .* mask),
+                parent(ocean_params.z0b .* (abs.(mask .- 1))),
+            )
+        dummmy_remap!(z0b_S, combined_field)
+    else
+        parent(combined_field) .=
+            combine_surface.(
+                parent(mask) .- parent(slab_ice_sim.integrator.p.ice_mask .* FT(2)),
+                parent(bucket_sim.integrator.u.bucket.T_sfc),
+                parent(slab_ocean_sim.integrator.u.T_sfc),
+                parent(slab_ice_sim.integrator.u.T_sfc),
+            )
+        dummmy_remap!(T_S, combined_field)
+        parent(combined_field) .=
+            combine_surface.(
+                parent(mask),
+                parent(bucket_sim.params.z_0m .* mask),
+                parent(slab_ocean_sim.integrator.p.params.z0m .* (abs.(mask .- 1))),
+            )
+        dummmy_remap!(z0m_S, combined_field)
+        parent(combined_field) .=
+            combine_surface.(
+                parent(mask),
+                parent(bucket_sim.params.z_0b .* mask),
+                parent(slab_ocean_sim.integrator.p.params.z0b .* (abs.(mask .- 1))),
+            )
+        dummmy_remap!(z0b_S, combined_field)
+        
+    end
+
+    # calculate turbulent fluxes on atmos grid and save in atmos cache
+    info_sfc = (; T_sfc = T_S, z0m = z0m_S, z0b = z0b_S, ice_mask = slab_ice_sim.integrator.p.ice_mask)
+    # This should also need q_sfc - currently it assumes q_sfc = q_sat
+    calculate_surface_fluxes_atmos_grid!(atmos_sim.integrator, info_sfc)
+end
+########
+# init coupling
+atmos_pull!(atmos_sim, slab_ice_sim, bucket_sim, slab_ocean_sim, mask, boundary_space, prescribed_sst, z0m_S,  z0b_S, T_S, ocean_params, SST)
+F_A, F_E, F_R, dF_A = atmos_push!(atmos_sim, boundary_space, F_A, F_E, F_R, dF_A, parsed_args)
+bucket_pull!(bucket_sim, F_A, F_E, F_R)
+reinit!(atmos_sim.integrator)
+reinit!(bucket_sim.integrator)
+if (prescribed_sst !== true) && (prescribed_sic == true)
+    ocean_pull!(slab_ocean_sim, F_A, F_R)
+    reinit!(slab_ocean_sim.integrator)
+end
+ice_pull!(slab_ice_sim, F_A, F_R, dF_A)
+reinit!(slab_ice_sim.integrator)
+
+if !is_distributed
+    check_conservation_callback(CS, atmos_sim, bucket_sim, F_A .+ F_R, F_E)
+end
+
+# At this stage, the integrators all have dY(0) computed based on stuff stored in aux, Y0, etc. - dE is up to date
+# Then we need to update aux to t1, because in the next step!, Y(1) will be computed based on Y0 and dY(0), and then
+# dY(1) will be computed using aux(1), Y(1), t(1)
+# But how to update aux to t(1) without Y(1)??
+@show "Starting coupling loop"
+walltime = @elapsed for t in (tspan[1]+Δt_cpl:Δt_cpl:tspan[end])
+    @show t
+    ## Atmos
+    atmos_pull!(atmos_sim, slab_ice_sim, bucket_sim, slab_ocean_sim, mask, boundary_space, prescribed_sst, z0m_S,  z0b_S, T_S, ocean_params, SST)
```

Comment:
thought that this line may need to be moved to *after* step!. step! does not update atmos_sim.integrator.p, atmos_pull! does. but we have already set atmos_sim.integrator.p for the first iteration in the atmos_pull!/reinit period above, before this loop starts?

this step is not computing anything new on the first iteration, I dont think.

Reply:
on the first step, the fluxes in line 218 are unchanged from the fluxes we computed up front.